### PR TITLE
Added Check to DelegateDialog to Alexa

### PR DESCRIPTION
**Pull request by jroodman**
This PR adds a check to DelegateDialog to Alexa for confirmation of user input. Also added is a new functionality - the ability for a user to clear all their items with a single Clear intent.

### CODE REVIEW

File: app/lib/handlers/intent_request_handlers/transaction_intent_request_handler.rb [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+      def slots
+        @slots ||= intent['slots'].to_h
+      end
+
+      def get_intent_specific_method(intent_name)
+        {
+          'Add'        => :process_add_items,
+          'Remove'     => :process_remove_items,
+          'Clear'      => :process_clear_items
```
Comment (rysospicy) [inline app/lib/handlers/intent_request_handlers/transaction_intent_request_handler.rb]: Extra space before the hash rockets here.  👾

File: app/lib/handlers/intent_request_handlers/transaction_intent_request_handler.rb [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+            user_id: user_id
+          )
+          if details[:time_til_warning] != 0
+            item[:warning_date] = Time.now + details[:time_til_warning].days
+          end
+          if details[:time_til_expiration] != 0
+            item[:expiration_date] = Time.now + details[:time_til_expiration].days
+          end
+          item.save
+          item
```
Comment (rysospicy) [inline app/lib/handlers/intent_request_handlers/transaction_intent_request_handler.rb]: I like to use `item.tap(&:save)` for things like this.  That syntax will call the method from the symbol on whatever's being tapped, then return the tapped thing.

**Review by rysospicy — APPROVED** [review]
👍

File: app/lib/handlers/intent_request_handlers/transaction_intent_request_handler.rb (hunk 2 above)
Comment (jroodman) [reply app/lib/handlers/intent_request_handlers/transaction_intent_request_handler.rb]: Fancy

**Pull request merged**